Use autosummary for sphinx docs

coretl opened the pull request:
This looks nicer in general and is less error prone.

Also fix an issue in the tests where pre-commit wasn't being run as files weren't added to git yet

callumforrester commented on tests/test_example.py:
Should it have its own test rather than be part of a big one?

coretl replied:
yes, done

coretl committed:
Use autosummary for sphinx docs

This looks nicer in general and is less error prone.

Also fix an issue in the tests where pre-commit wasn't being run as files
weren't added to git yet

coretl committed:
Add ADR

coretl commented:
> Does this need an ADR?

Yes. Done

callumforrester reviewed (approved):
LGTM
